browser/js/dashboard/newGraph/newGraph.controller.js

```diff
@@ -2,7 +2,10 @@ app.controller('newGraphCtrl', function ($scope, $q, WidgetSettingsFactory, Gene
 	$scope.getKeysAndTypes = function () {
 		DashboardFactory.getDataSource($scope.form.dataSource)
 		.then(DashboardFactory.findDataToGraph)
-		.then(validGraphFactory.getKeysAndTypes)
+		.then(function (realData) {
+           $scope.data = realData
```

Comment:
Is $scope.data used in the html anywhere? If not, just store it as a plain-old variable so that Angular doesn't have to add extra properties to the $scope.